src/test/clone/test_clone.c

```diff
@@ -52,15 +52,19 @@ static int _clone_minimal_thread(void* args) {
     _exit_thread(0);
 }
 
-static void _clone_minimal() {
-    // allocate some memory for the cloned thread.
-    uint8_t* stack = mmap(NULL, CLONE_TEST_STACK_NBYTES, PROT_READ | PROT_WRITE,
+static void _make_stack(void** top, void** bottom) {
+    *bottom = mmap(NULL, CLONE_TEST_STACK_NBYTES, PROT_READ | PROT_WRITE,
                           MAP_PRIVATE | MAP_ANONYMOUS | MAP_STACK, -1, 0);
-    assert_true_errno(stack != MAP_FAILED);
+    assert_true_errno(*bottom != MAP_FAILED);
+    *top = *bottom + CLONE_TEST_STACK_NBYTES;
 
-    // clone takes the "starting" address of the stack, which is the *top*.
-    uint8_t* stack_top = stack + CLONE_TEST_STACK_NBYTES;
+    // Set up a guard page
```

Comment:
Addition comments about why a guard page is needed would be useful to understand what is going on here.

Reply:
Done